src/main/java/net/backupcup/mcde/util/Choice.java

```diff
@@ -0,0 +1,48 @@
+package net.backupcup.mcde.util;
+
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+public class Choice {
+    private final SlotPosition choicePos;
+    private final EnchantmentSlot enchantmentSlot;
+
+    public Choice(EnchantmentSlot enchantmentSlot, SlotPosition choicePos) {
+        this.choicePos = choicePos;
+        this.enchantmentSlot = enchantmentSlot;
+    }
+
+    public int getLevel() {
+        return enchantmentSlot.getLevel();
+    }
+
+    public boolean isMaxedOut() {
+        return enchantmentSlot.isMaxedOut();
+    }
+
+    public boolean isChosen() {
+        return enchantmentSlot.getChosenPosition().map(choicePos::equals).orElse(false) &&
+            enchantmentSlot.getLevel() > 0;
+    }
+
+    public EnchantmentSlot getEnchantmentSlot() {
+        return enchantmentSlot;
+    }
+
+    public SlotPosition getChoicePosition() {
```

Comment:
Consider `getChoiceSlotPosition` or `getChoiceSlotPos`. Be sure that if you choose to use `Pos` over `Position`, that you are consistent throughout your code.